src/Microsoft.Extensions.FileProviders.Abstractions/NotFoundDirectoryContents.cs

```diff
@@ -13,6 +13,11 @@ namespace Microsoft.Extensions.FileProviders
     public class NotFoundDirectoryContents : IDirectoryContents
     {
         /// <summary>
+        /// A shared instance of <see cref="NotFoundDirectoryContents"/> 
+        /// </summary>
+        public static NotFoundDirectoryContents Singleton { get; } = new NotFoundDirectoryContents();
```

Comment:
We've use `Singleton` in abstractions, e.g. NullChangeToken.Singleton. Seems like they mean the same thing. Is there some nuance I'm missing?